front/pages/w/[wId]/u/chat/[cId]/index.tsx

```diff
@@ -919,7 +965,46 @@ export default function AppChat({
     })();
     setLoading(false);
   };
+  const handleFeedback = (
+    message: ChatMessageType,
+    feedback: MessageFeedbackStatus
+  ) => {
+    const messagesPlusFeedback = messages.map((m) => {
+      if (m === message) {
+        if (m.feedback !== feedback) {
+          m.feedback = feedback;
+        } else {
+          m.feedback = null;
+        }
+      }
+      return m;
+    });
+    setMessages(messagesPlusFeedback);
+
+    // if the feedback was given on the latest message a few seconds after it
+    // finished (potentially common use case), said latest message might not yet
+    // have been persisted in the database and thus not have an id. This is why
+    // we use storeChatSession
+    void storeChatSession(title, messagesPlusFeedback);
```

Comment:
I understand the rationale here, but resending the whole chatSession is hella inefficient. And I admit that the whole session storage approach is quite inefficient. I thiiink we want to maybe fix a few things here so that we can get into a world where we can update a single message?